elytron/src/main/java/org/wildfly/extension/elytron/ElytronSubsystemTransformers.java

```diff
@@ -153,6 +158,28 @@ private static void from17(ChainedTransformationDescriptionBuilder chainedBuilde
                 .getAttributeBuilder()
                 .addRejectCheck(RejectAttributeChecker.DEFINED, TRUSTED_VIRTUAL_SECURITY_DOMAINS)
                 .setDiscard(DiscardAttributeChecker.UNDEFINED, TRUSTED_VIRTUAL_SECURITY_DOMAINS);
+
+        builder.addChildResource(PathElement.pathElement(MAPPED_ROLE_MAPPER))
+                .getAttributeBuilder()
+                .setDiscard(DiscardAttributeChecker.UNDEFINED, FROM)
+                .setDiscard(DiscardAttributeChecker.UNDEFINED, TO)
+                .addRejectCheck(RejectAttributeChecker.DEFINED, FROM)
+                .addRejectCheck(RejectAttributeChecker.DEFINED, TO)
+                .addRejectCheck(new RejectAttributeChecker.DefaultRejectAttributeChecker() {
+                    /* [WFCORE-6244] role-map attribute was previously formatted (directly) as a string to string-list mapping of roles.
+                     * Attributes formatted as a list of maps (ObjectListAttributeDefinition) should be rejected. */
+                    @Override
+                    protected boolean rejectAttribute(PathAddress address, String attributeName, ModelNode attributeValue, TransformationContext context) {
+                        return attributeValue.getType() != ModelType.OBJECT;
+                    }
+
+                    @Override
+                    public String getRejectionLogMessage(Map<String, ModelNode> attributes) {
+                        return ROOT_LOGGER.mappedRoleMapExpectsObject(
+                                attributes.get("role-map") != null ? attributes.get("role-map").getType().toString() : "not")
```

Comment:
We shouldn't hardcode the word "not" here (all message strings should be in the messages class. Maybe the error message could be made more generic, perhaps something along the lines of:

"Value for attribute `role-map` is invalid"